### ISSUE TITLE
for azurerm_load_test identity - update documentation - input a block not a string & extend to allow UserAssigned

### ISSUE THREAD

**Issue by sharccc** [hashicorp/pandora#2270]
documentation is currently missing some readability details - input for `identity` should be a block containing type.
> https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/load_test

currently:
* `identity` - (Optional) Specifies the Managed Identity which should be assigned to this Load Test.

should be:
```
* `identity` - (Optional) An `identity` block as defined below.
---
An `identity` block supports the following:
* `type` - (Required) The type of Managed Identity used for this Load Test. Possible values are `SystemAssigned`.
```

also
requesting the identity aligns with AzureRM template and accepts both types of Managed Identities

> `type` | Type of managed service identity (where both SystemAssigned and UserAssigned types are allowed).
> `identity_ids` | The set of user assigned identities associated with the resource. etc
> https://learn.microsoft.com/en-us/azure/templates/microsoft.loadtestservice/loadtests?pivots=deployment-language-terraform

(as an example)
There is a similar identity block available for AzureRM Automation Account - allowing 3 options:

> An identity block supports the following:
> `type` - (Required) The type of identity used for this Automation Account. Possible values are SystemAssigned, UserAssigned and SystemAssigned, UserAssigned.
> `identity_ids`- (Optional) The ID of the User Assigned Identity which should be assigned to this Automation Account.
> Note:
> identity_ids is required when type is set to UserAssigned or SystemAssigned, UserAssigned.
> https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/automation_account

support case 104926 
also feature request 1204255964099438 

thanks - Sharon (HashiCorp Terraform Support)

**Comment by sharccc** [comment]
also feature request 1204255964099438
requesting the `identity` aligns with AzureRM template and accepts both types of Managed Identities

> type Type of managed service identity (where both SystemAssigned and UserAssigned types are allowed).
> https://learn.microsoft.com/en-us/azure/templates/microsoft.loadtestservice/loadtests?pivots=deployment-language-terraform

There is a similar `identity` block available for AzureRM Automation Account - allowing 3 options:
> `values are SystemAssigned, UserAssigned and SystemAssigned, UserAssigned.`
> https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/automation_account

**Comment by sharccc** [comment]
Looks like 

`"identity": commonschema.SystemAssignedIdentityOptional(), `
[load_test_resource_gen.go#L61](https://github.com/hashicorp/terraform-provider-azurerm/blob/d7dc52d8a41e1e2fc90d683cfb8f8f5cc1294f9c/internal/services/loadtestservice/load_test_resource_gen.go#L61)

The current schema has validation accepting 'SystemAssigned'
commonschema.SystemAssignedIdentityOptional()
  ValidateFunc: validation.StringInSlice([]string{   
    string(identity.TypeSystemAssigned),
  },
[SystemAssignedIdentityOptional](https://github.com/hashicorp/go-azure-helpers/blob/main/resourcemanager/commonschema/identity_system.go#:~:text=func%20SystemAssignedIdentityOptional-,(,-)%20*schema)

There is an alternate schema allowing either 'SystemAssigned' or 'UserAssigned'
commonschema.SystemOrUserAssignedIdentityOptional()
  ValidateFunc: validation.StringInSlice([]string{   
    string(identity.TypeUserAssigned), 
    string(identity.TypeSystemAssigned),
  }
[SystemOrUserAssignedIdentityOptional](https://github.com/hashicorp/go-azure-helpers/blob/main/resourcemanager/commonschema/identity_system_or_user.go#:~:text=func-,SystemOrUserAssignedIdentityOptional,-()%20*schema)

**Comment by tombuildsstuff** [comment]
Digging into this one, this appears to be a larger issue in that we're not outputting the documentation for nested blocks, in addition to misidentifying `identity` as a field rather than a block - will open a PR to fix both, which'll fix this issue.

**Comment by tombuildsstuff** [comment]
FWIW support for User Assigned Identities is available in the next API Version, which there's a few remaining items to unblock, I'm going to re-focus this issue on the docs side and I've split out a separate issue for updating the API Version here (for which there's a few blockers): https://github.com/hashicorp/pandora/issues/2549
